tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModule2.fs

```diff
@@ -369,287 +369,296 @@ type OperatorsModule2() =
         
         ()
     
     [<Test>]
     member this.snd() = 
         // value type        
         let result = Operators.snd ("ABC",100)
-        Assert.AreEqual(result,100) 
+        Assert.AreEqual(100, result) 
         
         // reference type        
         let result = Operators.snd (100,"ABC")
-        Assert.AreEqual(result,"ABC") 
+        Assert.AreEqual("ABC", result) 
         
         // null        
         let result = Operators.snd (100,null)
-        Assert.AreEqual(result,null) 
+        Assert.AreEqual(null, result) 
         
         ()
     
     [<Test>]
     member this.sqrt() = 
         // double        
         let result = Operators.sqrt 100.0
-        Assert.AreEqual(result,10) 
+        Assert.AreEqual(10.0, result) 
         
         ()
     
     [<Test>]
     member this.stderr() =         
         let result = Operators.stderr 
-        Assert.AreEqual(result.WriteLine("go"),null) 
+        Assert.AreEqual(null, result.WriteLine("go")) 
         
         ()
     
     [<Test>]
     member this.stdin() =         
         let result = Operators.stdin 
-        Assert.AreEqual(result.Dispose(),null)
+        Assert.AreEqual(null, result.Dispose())
         
         ()   
     
     [<Test>]
     member this.stdout() =         
         let result = Operators.stdout 
-        Assert.AreEqual(result.WriteLine("go"),null)
+        Assert.AreEqual(null, result.WriteLine("go"))
         
         ()   
     
     [<Test>]
     member this.string() =  
         // value type
         let result = Operators.string 100
-        Assert.AreEqual(result,"100")
+        Assert.AreEqual("100", result)
         
         // reference type
         let result = Operators.string "ABC"
-        Assert.AreEqual(result,"ABC")
-        
-        // unit
-        CheckThrowsNullRefException(fun () -> Operators.string null |>ignore)
+        Assert.AreEqual("ABC", result)
         
         ()      
     
     [<Test>]
     member this.tan() =  
         // double
         let result = Operators.tan 1.0
-        Assert.AreEqual(result.ToString(),"1.5574077246549")
+        Assert.AreEqual(1.5574077246549023, result)
         
         ()    
     
     [<Test>]
     member this.tanh() =  
-        // double
-        let result = Operators.tanh 0.8
-        Assert.AreEqual(result,0.664036770267849)
+        // this doesn't work, even though the values are roundtrippable correct
+        // it also works correctly in FSI, so what gives?
+        //let result = Operators.tanh 0.8
+        //Assert.AreEqual(0.66403677026784891, result)
         
+        let result = Operators.tanh 0.8
+        // why does the following not compile?
+        //Assert.IsTrue(result = 0.66403677026784891)
+        // but the following does?
+        // Assert.IsTrue(0.5 = 0.66403677026784891)
+        // this works in FSI, but fails just like Assert.AreEqual, what is wrong here?
+        Assert.IsTrue(result.Equals(0.66403677026784891))
```

Comment:
@KevinRansom this looks fun

Reply:
@cartermp, @KevinRansom, part of this reported separately just now: https://github.com/dotnet/fsharp/issues/9517

Reply:
Note that `Assert.AreEqual(0.66403677026784891, result)` throws an assertion failure. But it doesn't do that when run (with NUnit, same version, loaded) in FSI. And it shouldn't, because, well, the number is correct ;). Same story for the other examples, so there may be something fishy going on here (instead of just a significant digit rounding error issue).

Reply:
@cartermp & @KevinRansom, this is now resolved, and new issues created where it wasn't. I've decided to make it abundantly clear that there's a difference here in x86 and x64 behavior that's outside of F#'s control.